### PR TITLE
color.ts: code cleanup

### PR DESCRIPTION
**Pull request by subzey**
**Type of PR:** enhancement

**PR checklist:**

- [ ] Addresses an existing issue: fixes #🐱
- [x] Includes tests
- [ ] Documentation update

**Overview of change:**

Removed buncha unused and overcompilcated code.
Result is -125 B min.gz and much more restricted module exports.

**Is there anything you'd like reviewers to focus on?**

Each change has its own commit, you probably find it easier to review commit-wise

### CODE REVIEW

File: src/helpers/color.ts [hunk 1]
```diff
@@ -346,11 +330,12 @@ export function parseRgb(colorString: string): Rgb {
...
-const rgbGrayscaleWeights = [0.199, 0.687, 0.114];
-export function rgbToGrayscale(rgbValue: Rgb): number {
-	return rgbGrayscaleWeights[0] * rgbValue[0] +
-		rgbGrayscaleWeights[1] * rgbValue[1] +
-		rgbGrayscaleWeights[2] * rgbValue[2];
+function rgbToGrayscale(rgbValue: Rgb): number {
+	return (
+		0.199 * rgbValue[0] +
+		0.687 * rgbValue[1] +
+		0.114 * rgbValue[2]
```
Comment (timocov) [inline src/helpers/color.ts]: I would like to have that constants (0.199, 0.687, 0.114) with their names. You can use `const enum RgbGrayscaleWeights {}` for example, in local constants (I believe they should be inline as well).

File: src/helpers/color.ts [hunk 2]
```diff
@@ -197,19 +193,11 @@ function normalizeNumber(min: number, n: number, max: number): number {
 	);
 }
 
-export function normalizeRedComponent(red: number): RedComponent {
-	return normalizeInteger(0, red, 255) as RedComponent;
+function normalizeColorComponent<T extends RedComponent | GreenComponent | BlueComponent>(component: number): T {
```
Comment (Nipheris) [inline src/helpers/color.ts]: `normalizeRgbComponent`? Different components in different color models are normalized to other ranges. Even alpha is normalized to [0, 100].

File: src/helpers/color.ts (hunk 1 above)
Comment (subzey) [reply src/helpers/color.ts]: @timocov Weights `0.199`, `0.687`, `0.114` are probably a typo.

https://en.wikipedia.org/wiki/YUV#Y.E2.80.B2UV444_to_RGB888_conversion mentions `0.299`, `0.587`, `0.114`

Comment (subzey) [reply src/helpers/color.ts]: Fixed, incl coefficients

File: src/helpers/color.ts (hunk 2 above)
Comment (subzey) [reply src/helpers/color.ts]: Changed

File: src/helpers/color.ts (hunk 1 above)
Comment (subzey) [reply src/helpers/color.ts]: Reverted coefficients change.

File: src/helpers/color.ts [hunk 3]
```diff
@@ -346,11 +330,17 @@ export function parseRgb(colorString: string): Rgb {
 	}
 }
 
-const rgbGrayscaleWeights = [0.199, 0.687, 0.114];
-export function rgbToGrayscale(rgbValue: Rgb): number {
-	return rgbGrayscaleWeights[0] * rgbValue[0] +
-		rgbGrayscaleWeights[1] * rgbValue[1] +
-		rgbGrayscaleWeights[2] * rgbValue[2];
+function rgbToGrayscale(rgbValue: Rgb): number {
+	// See https://en.wikipedia.org/wiki/YUV#Y.E2.80.B2UV444_to_RGB888_conversion
```
Comment (timocov) [inline src/helpers/color.ts]: It looks like the values from this link aren't the same as below ones. In this case I would prefer to change this comment to something like:

```
// the following values are from https://en.wikipedia.org/wiki/YUV#Y.E2.80.B2UV444_to_RGB888_conversion
// but a little bit adjusted for real use cases
```

Comment (subzey) [reply src/helpers/color.ts]: check

**Comment by subzey** [comment]
Here comes the tests. Partially based on @Nipheris' code

**Pull request merged**